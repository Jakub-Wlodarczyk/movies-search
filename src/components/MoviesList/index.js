import React from 'react';
import noImage from '../../assets/sample_poster.jpg';
import { getMovieDetails } from '../../app/resources';
import MovieDetails from '../MovieDetails';

/**
 * The function displays fetched results as a grid
 * @param {Array} movies
 * @param {String} baseUrl
 * @returns {JSX}
 */
class MoviesList extends React.Component {
	constructor (props) {
		super(props);

		this.state = {
			showMoviesList: true,
			showMovieDetails: false,
			movieDetails: {}
		};

		this.getMovieDetails = this.getMovieDetails.bind(this);
	}

	getMovieDetails (e) {
		const movieId = e.target.id;

		getMovieDetails(movieId).then(details => this.setState({
			movieDetails: details,
			showMoviesList: false,
			showMovieDetails: true
		}));
	}

	render () {
		const { movies, baseUrl } = this.props;

		return (
			<div>
				{this.state.showMoviesList &&
				<div className="movies-list">
					{movies.map(movie => (
						<div key={`${movie.title}-${movie.id}`} className="movies-list__single-movie">
							<p>TITLE: {movie.title}</p>
							<p>RELEASE DATE: {movie.release_date}</p>
							<img
								name={`${movie.title}-image`}
								src={movie.poster_path === null ? noImage : `${baseUrl}w500/${movie.poster_path}`}
								id={movie.id}
								onClick={this.getMovieDetails}
							/>
						</div>
					))}
				</div>
				}
				{this.state.showMovieDetails && <MovieDetails baseUrl={baseUrl} details={this.state.movieDetails} />}
			</div>
		);
	}
}

MoviesList.propTypes = {
	movies: React.PropTypes.arrayOf(React.PropTypes.object).isRequired,
	baseUrl: React.PropTypes.string.isRequired
};

export default MoviesList;
