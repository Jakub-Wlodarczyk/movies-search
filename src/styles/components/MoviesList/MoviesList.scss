.movies-list {
  display: flex;
  justify-content: center;
  flex-wrap: wrap;
  width: 100%;

  .movies-list__single-movie {
    width: 32%;
    margin-right: 1%;
    text-align: center;

    img {
      width: 100%;
      height: auto;
      cursor: pointer;
    }
  }

  .movies-list__single-movie:nth-child(3n) {
    margin-right: 0;
  }
}
