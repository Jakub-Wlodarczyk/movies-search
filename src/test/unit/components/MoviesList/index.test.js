import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import MoviesList from '../../../../components/MoviesList';

describe('MoviesList', () => {
	it('should render the MoviesList view', () => {
		const props = {
			movies: [
				{
					title: 'title 1',
					release_date: '01-01-2000',
					poster_path: '/path1'
				},
				{
					title: 'title 2',
					release_date: '02-02-2000',
					poster_path: '/path2'
				}
			],
			baseUrl: '/some-base-url'
		};

		const wrapper = shallow(<MoviesList {...props} />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it('should render the MoviesList container', () => {
		const props = {
			movies: [
				{
					title: 'title 1',
					release_date: '01-01-2000',
					poster_path: '/path1'
				},
				{
					title: 'title 2',
					release_date: '02-02-2000',
					poster_path: '/path2'
				}
			],
			baseUrl: '/some-base-url'
		};

		const wrapper = shallow(<MoviesList {...props} />);
		expect(wrapper.find('.movies-list').exists()).toBeTruthy();
	});
});
