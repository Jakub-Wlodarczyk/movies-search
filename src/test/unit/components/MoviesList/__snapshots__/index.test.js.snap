// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesList should render the MoviesList view 1`] = `
<div>
  <div
    className="movies-list"
  >
    <div
      className="movies-list__single-movie"
    >
      <p>
        TITLE: 
        title 1
      </p>
      <p>
        RELEASE DATE: 
        01-01-2000
      </p>
      <img
        name="title 1-image"
        onClick={[Function]}
        src="/some-base-urlw500//path1"
      />
    </div>
    <div
      className="movies-list__single-movie"
    >
      <p>
        TITLE: 
        title 2
      </p>
      <p>
        RELEASE DATE: 
        02-02-2000
      </p>
      <img
        name="title 2-image"
        onClick={[Function]}
        src="/some-base-urlw500//path2"
      />
    </div>
  </div>
</div>
`;
